apps/wallet-mobile/src/features/Receive/useCases/ReceiveScreen.tsx

```diff
@@ -0,0 +1,126 @@
+import {useFocusEffect} from '@react-navigation/native'
+import {useTheme} from '@yoroi/theme'
+import _ from 'lodash'
+import React from 'react'
+import {SafeAreaView, ScrollView, StyleSheet, View} from 'react-native'
+
+import Icon from '../../../assets/img/copy.png'
+import {Button, Spacer, StatusBar} from '../../../components'
+import {useCopy} from '../../../legacy/useCopy'
+import {useMetrics} from '../../../metrics/metricsManager'
+import {useSelectedWallet} from '../../../SelectedWallet'
+import {useHideBottomTabBar, useReceiveAddresses} from '../../../yoroi-wallets/hooks'
+import {AddressDetailCard} from '../common/AddressDetailCard/AddressDetailCard'
+import {mocks as mockReceives} from '../common/mocks'
+import {SkeletonAdressDetail} from '../common/SkeletonAddressDetail/SkeletonAddressDetail'
+import {useNavigateTo} from '../common/useNavigateTo'
+import {useStrings} from '../common/useStrings'
+
+export const ReceiveScreen = () => {
+  useHideBottomTabBar()
+  const strings = useStrings()
+  const {styles, colors} = useStyles()
+  const wallet = useSelectedWallet()
+  const receiveAddresses = useReceiveAddresses(wallet)
+  const navigate = useNavigateTo()
+
+  const [isCopying, copy] = useCopy()
+
+  const currentAddress = _.last(receiveAddresses)
+
+  React.useEffect(() => {
+    wallet.generateNewReceiveAddressIfNeeded()
+  }, [wallet])
+
+  const {track} = useMetrics()
+
+  useFocusEffect(
+    React.useCallback(() => {
+      track.receivePageViewed()
+    }, [track]),
+  )
+
+  const onRequestSpecificAmount = () => {
+    navigate.specificAmount()
+  }
+
+  return (
+    <SafeAreaView style={styles.root}>
```

Comment:
SafeAreView must have the edges set (if has navbottom bottom is not required), --> unless it was mounted on navigator, and it so, it can be replaced by view.